### ISSUE TITLE
Serious lag when playing audio

### ISSUE THREAD

**Issue by Cat7373** [iina/iina#4049]
**System and IINA version:**

- macOS 13.0
- IINA 1.3.1

**Expected behavior:**
1. open the audio file
2. Quickly start playing
3. Click on the progress bar to quickly finish skipping

**Actual behavior:**
1. open the audio file
2. Pause for a while and start playing
3. Click on the progress bar, lag for a while, finish skipping

**Steps to reproduce:**
1. Play [this audio file](https://drive.google.com/file/d/1OOM9u2VNIg2u6P-Y4WADO4zi93pN02Id/view?usp=sharing)
2. View [this demo file](https://drive.google.com/file/d/19ND6szqDLmao3Ephm2BagN1jVhrF8LVH/view?usp=sharing)

- [ ] MPV does not have this problem.

**How often does this happen?**
each time.

**Comment by low-batt** [comment]
It is working fine for me a lot of the time. So much so I initially thought I could not reproduce a problem. And then it glitched on me. Not quite the same as I see in the demo file. Audio continued but the knob in the slider stopped updating. Something is definitely wrong.

**Comment by Cydiater** [comment]
Similar to #3635. I posted it earlier, but no one responded.

**Comment by low-batt** [comment]
I have tracked down the problem and will be proposing a fix. This is a regression in 1.3.1. I've replied to @Cydiater in issue #3635. Although that issue is similar to this one, I suspect something else is causing that issue.

**Comment by low-batt** [comment]
I was thinking the cause of this issue was the same as the cause of issue #4051. The cause of that issue is a regression involving updating the progress bar. After reviewing the video demonstrating the problem again I think I did not reproduce the problem being reported in this issue. I'm now thinking @Cydiater may be correct and this issue is related to #3635 or something similar.

Do you have any interesting audio/video applications installed? Especially applications from [Rogue Amoeba](https://www.rogueamoeba.com/)?

Are you using any external audio devices?

Open [Console](https://support.apple.com/en-in/guide/console/welcome/mac). Click on `Crash Reports`. Do you see any reports for macOS daemons such as `coreaudiod`?

**Comment by Cat7373** [comment]
Strangely, when I tried to reproduce it, the lag disappeared.

**Comment by Cat7373** [comment]
I found out that the problem only occurs when I connect a bluetooth headset, my headset is a Sony WI-SP510, I don't see any messages in the Crash Reports in Console

**Comment by Cat7373** [comment]
This issue does not occur when hitting space to preview audio files

**Comment by Cat7373** [comment]
Turning off hardware decoding does not solve this problem

**Comment by low-batt** [comment]
That the problem only occurs when using a Bluetooth headset confirms @Cydiater is correct and this is not the IINA UI regression that I hit while trying to reproduce this one.

IINA uses a library from [mpv](https://mpv.io/) for audio/video playback. This is the reason for the question in the issue template about whether the mpv player also exhibits the problem. If it is an audio or video playback issue it is likely that the root cause of the problem is in the library IINA embeds from the mpv project, or some library the mpv library depends upon. 

The problem here appears to be the open mpv issue [#10270](https://github.com/mpv-player/mpv/issues/10270) that reports lagging when using AirPods. Although that issue is focused on AirPods, the mpv issue [#9243](https://github.com/mpv-player/mpv/issues/9243) (closed as a duplicate) indicates the problem is with AirPlay and Bluetooth and also will be encountered with Bluetooth headsets.

Such problems are referred to as "upstream" as they have to be fixed by another project and then IINA will need to pickup the fix once that project has released a new version.

So for now we are waiting on a fix from the mpv project.

**Comment by low-batt** [comment]
The mpv issue [#10270](https://github.com/mpv-player/mpv/issues/10270) has been closed because a fix has been merged into the mpv master branch. We are now waiting on the next mpv release.

**Comment by Cat7373** [comment]
:tada: :tada: :tada:

**Comment by low-batt** [comment]
As you can see from the PR referenced above the next release of IINA will use a newer version of mpv that contains a fix for this issue.

**Comment by Cat7373** [comment]
In IINA 1.3.1 (build 133, mvp 0.34.1) and macOS 13.2, the problem does not recur

**Comment by low-batt** [comment]
This is fixed in the released version of IINA.
